dataset/include/scipp/dataset/dataset.h

```diff
@@ -208,6 +211,19 @@ copy(const Dataset &dataset, Dataset &out,
 copy(const Dataset &dataset, Dataset &&out,
      const AttrPolicy attrPolicy = AttrPolicy::Keep);
 
+[[maybe_unused]] SCIPP_DATASET_EXPORT Dataset &
+assign_from(Dataset &, const Dataset &other);
```

Comment:
Would it be nicer to have this as a member?

Reply:
it might be a bit more natural to use. But it would close off the set of potential types for `other`. Having this as a free function allows users to add overloads for other containers. But I don't know if this matters at all.

Reply:
Valid point about ADL, however if this is not required - I find the argument about it being more natural to have as a member more compelling. If would seem to me to best sit alongside copy/move related member functions.

Reply:
I thought the same as @OwenArnold, but when a started making the change of functions to members I realized that we cannot generate the functions (for data arrays) then, so I stopped for now, not knowing what to do.

Reply:
I am getting confused about masks now: Should this really behave like, e.g., `operator+=` which ORs the masks? Wouldn't that be weird? Shouldn't it instead replace/copy the masks?

Reply:
Consider:
```python
da1['x', 4] = da2
```
- I would expect existing masks to be replaced by masks of `da2`.
- If `da1` has masks that are not in `da2` should this fail? We cannot remove the mask when operating on a slice.
- If `da2` has masks that are not in `da1` should this fail? We cannot add the mask when operating on a slice.